File: module_fragment_02/src/main/java/cn/jack/module_fragment_02/mvvm/SubjectViewModel.java
package cn.jack.module_fragment_02.mvvm;

import android.app.Application;
import androidx.annotation.NonNull;
import javax.inject.Inject;
import cn.jack.library_common_business.entiy.ProjectInfoList;
import cn.jack.library_util.LogUtils;
import cn.jack.module_fragment_02.contract.ISubjectLisenter;
//import cn.jack.module_fragment_02.di.DaggerViewModelComponent;
import jack.retrofit2_rxjava2.exception.ErrorStatusInfo;
import jack.wrapper.base.mvvm.viewModel.BaseViewModel;
import jack.wrapper.base.mvvm.viewModel.liveData.UIChangeLiveData;

/**
 * @创建者 Jack
 * @创建时间 2021/3/30 16:56
 * @描述
 */
public class SubjectViewModel extends BaseViewModel<SubjectHttpRepository> implements ISubjectLisenter {

    @Inject
    SubjectHttpRepository mSubjectHttpRepository;

    public UIChangeLiveData<ProjectInfoList> mProjectInfoListUIChangeLiveData = new UIChangeLiveData<>();
    public UIChangeLiveData<String> mOnfailedUIChangeLiveData = new UIChangeLiveData<>();

    public SubjectViewModel(@NonNull Application application) {
        super(application);

        LogUtils.d("初始化实际 " + System.currentTimeMillis());
//        DaggerViewModelComponent.builder().build().inject(this);
        mModel = mSubjectHttpRepository;
    }

    public void listSubject(boolean refresh,String articleId) {
        mModel.pageSubject(refresh,articleId,this);
    }

    @Override
    public void onSuccess(ProjectInfoList projectInfoList) {
        mProjectInfoListUIChangeLiveData.setValue(projectInfoList);
    }

    @Override
    public void loadFailed(ErrorStatusInfo errorStatusInfo) {
        super.loadFailed(errorStatusInfo);
        mOnfailedUIChangeLiveData.setValue("");
    }

}

File: module_fragment_02/src/main/java/cn/jack/module_fragment_02/mvvm/SubjectViewModel.kt
package cn.jack.module_fragment_02.mvvm

import androidx.lifecycle.viewModelScope
import cn.jack.library_common_business.entiy.ProjectInfoList
import com.jack.lib_wrapper_mvvm.base.viewmodel.BaseWrapperViewModel
import com.jack.lib_wrapper_net.model.EventResult
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.*
import javax.inject.Inject

/**
 * @创建者 Jack
 * @创建时间 2021/3/30 16:56
 * @描述
 */
@HiltViewModel
class SubjectViewModel @Inject constructor(private val mRepository: SubjectHttpRepository) :
    BaseWrapperViewModel() {

    private val mProjectInfoList =
        MutableStateFlow<EventResult<ProjectInfoList>>(EventResult.OnComplete)

    val projectInfoList_ = mProjectInfoList.shareIn(viewModelScope, SharingStarted.WhileSubscribed(5000))

    fun listSubject(refresh: Boolean, articleId: String) {
        mRepository.pageSubject(refresh, articleId).onEach {
            mProjectInfoList.value = it
        }.launchIn(viewModelScope)
    }

}